### ISSUE TITLE
uBlock appears to be unable to update filters in Firefox's private mode

### ISSUE THREAD

**Issue by UltramarineBlue0** [uBlockOrigin/uBlock-issues#409]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

While uBO can't access indexedDB in Firefox's permanent private mode is a known issue, it *appears* that it can't update the filters in private mode as well. Updating using the "Update now" button doesn't change the filter count. (I'm not sure if it's a display issue or if no filters are actually loaded)

### A specific URL where the issue occurs

[A specific URL is MANDATORY for issue happening on a web page, even if it happens "everywhere"]

### Steps to Reproduce

1. Start Firefox normally (not in private mode) and make sure the filters are up to date
2. Close Firefox then restart Firefox with the command line options: '-no-remote -private'
3. Go to uBlock Origin Setting and click "Update now"

### Expected behavior:

The filters are initially all out of date. After updating the filters, the yellow triangles disappear and the filter count changes because newer version of the filters are loaded. The filter count should be roughly equal the filter count in Firefox normal mode (before the restart with extra command line options)

### Actual behavior:

The filters are initially all out of date. After updating the filters, the yellow triangles disappear. However neither the network filter count nor the cosmetic filter count change, giving the appearance as if none of the filters are updated. The filter count is also significantly lower than in Firefox normal mode (before the restart with extra command line options)

### Your environment

* uBlock Origin version: 1.18.2 (however this is happening for at least a few months)
* Browser Name and version: Firefox 60.5.0 ESR on Win10 && Firefox 64 on Linux
* Operating System and version: Windows 10 Update 1803 && Lubuntu 18.04.1

**Comment by uBlock-user** [comment]
Duplicate of https://github.com/gorhill/uBlock/issues/3071

**Comment by uBlock-user** [comment]
> Is this the expected behavior?

Yes, part of the issue amount others(like storage appearing 0 bytes.)

**Comment by gorhill** [comment]
uBO is not allowed to save the lists it downloads, and as a result always ends up loading the list from the package when it tries to reload all the filter lists after the lists have been "updated".

**Comment by UltramarineBlue0** [comment]
Is it reasonable / feasible to add a workaround for private mode that only loads the updated filters into memory without trying to save them?

**Comment by gorhill** [comment]
Caching in memory sounds abusive.

Another idea -- which I entertained when I fixed <https://github.com/uBlockOrigin/uBlock-issues/issues/399> -- is to allow uBO to fallback using `browser.storage.local` on Firefox for caching purpose, just like it is with Chromium. There is a new hidden setting which allows a user to force a specific storage implementation for cache purpose, currently looked up only on non-Firefox browsers. I could add support for a `webext` value, to force uBO to use `browser.storage.local`. My understanding is that this storage is not affected by private mode.
